### PR TITLE
Close image rawSource when each loop ends

### LINKED ISSUE #8583: (podman image sign) uses an ImageSource after closing it
After https://github.com/containers/podman/blob/70284b18cc7d4dc478c439cdda60ba64cfec060b/pkg/domain/infra/abi/images.go#L725 , all future uses of `rawSource` are invalid.

### PR DESCRIPTION
**Pull request by QiWang19**
Previously close rawSouce in the middle makes future use of rawSource invalid.
Move the rawSource.Close() to the end of each loop.

Close #8583 

Signed-off-by: Qi Wang <qiwan@redhat.com>

### CODE REVIEW

**Comment by QiWang19** [comment]
@mtrmac PTAL

**Review by mtrmac — COMMENTED** [review]
Clearly an improvement, thanks for tackling this!

Ideally the loop body would be moved into a separate function (or an internal anonymous function, `func(){}()` and `defer` would be used to ensure the source is closed on all error paths.

File: pkg/domain/infra/abi/images.go [hunk 1]
```diff
@@ -722,10 +722,6 @@ func (ir *ImageEngine) Sign(ctx context.Context, names []string, options entitie
 		if err != nil {
 			return nil, errors.Wrapf(err, "error getting image source")
 		}
-		err = rawSource.Close()
```
Comment (rhatdan) [inline pkg/domain/infra/abi/images.go:725]: There is a chance at rawsouce not being closed.  I agree with Miloslav, to move this to a function and 
defer rawSource.Close()

**Comment by rhatdan** [comment]
LGTM, although I would have created a separate named function, but this works.

**Review by mtrmac — APPROVED** [review]
LGTM.

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Comment by TomSweeneyRedHat** [comment]
/lgtm
/hold
holding for tests

**Comment by QiWang19** [comment]
Needs approved label.

**Pull request merged**